src/App.test.js

```diff
@@ -1,9 +1,90 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
-
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+const puppeteer = require('puppeteer');
+const faker = require('faker');
+
+// Uses faker to set inital "person" object, This can be used whenever you need to fill in a form with any param inside the object.
+const person = {
+  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
+  email: "c@lvin.com",
+  phone: faker.phone.phoneNumber(),
+  message: faker.random.words(),
+  password: "password"
+};
+
+
+// Inital setup for pupeteer
+let browser;
+let page;
+const width = 1920;
+const height = 1080;
+
+// Setup before any test is run, launches puppeteer with the window sizes & heigh
+beforeAll(async () => {
+  browser = await puppeteer.launch({
+    headless: true,
+    slowMo: 80,
+    args: [`--window-size=${width},${height}`]
+  });
+  page = await browser.newPage();
+  await page.setViewport({ width, height });
+});
+// after all tests are complete close the browser
+afterAll(() => {
+  browser.close();
 });
+
+
+
+const appUrlBase = 'http://localhost:3000'
+const routes = {
+  public: {
+    login: `${appUrlBase}/login`,
+    noMatch: `${appUrlBase}/PageNotFound`,
+  },
+  private: {
+    dashboard: `${appUrlBase}/`,
+    registration: `${appUrlBase}/registration`,
+  },
+};
+
+describe('Signing In', () => {
+  test('sign-in title loads correctly', async () => {
+
+    await page.goto(routes.public.login);
+    await page.waitForSelector('.header');
+
+    const html = await page.$eval('.header', e => e.innerHTML);
+    expect(html).toBe('sign in');
+
+
+  }, 160000000);
+  test('login form leads to the dashboard', async () => {
+
+    // you will need to find the exact path for the screen shots to save in the file. 
+
+    await page.screenshot({ path: "debug_screenshots/test.png" })
+    // Uses faker to load data from object uptop
+    await page.goto(routes.public.login);
+    await page.click('.email')
+    await page.type('.email', person.email)
+    await page.click('.password')
+    await page.type('.password', person.password)
+    await page.click('.remember')
+    await page.click('.signIn')
+
```

Comment:
This would most likely need a `page.wait()` when `slowMo` is removed.

Reply:
I took out `slowMo` and the test finish in 5 secs. Where do i add page.wait()

Reply:
After you click sign in, but before you check the URL (give time for login to be validated and the redirect to occur).